worker/uniter/resolver/loop.go

```diff
@@ -60,16 +60,44 @@ func Loop(cfg LoopConfig, localState *LocalState) error {
 		return errors.Trace(err)
 	}
 
+	fire := make(chan struct{}, 1)
 	for {
 		rf.RemoteState = cfg.Watcher.Snapshot()
 		rf.LocalState.State = cfg.Executor.State()
 
 		op, err := cfg.Resolver.NextOp(*rf.LocalState, rf.RemoteState, rf)
 		for err == nil {
+			// Send remote state changes to running operations.
+			remoteStateChanged := make(chan remotestate.Snapshot)
+			done := make(chan struct{})
+			go func() {
+				var rs chan remotestate.Snapshot
+				for {
+					select {
+					case <-cfg.Watcher.RemoteStateChanged():
+						// We consumed a remote state change event
+						// so we need a way to trigger the select below
+						// in case it was a new operation.
```

Comment:
Do we have a unit test for this?

Reply:
LoopSuite.TestLoopWithChange

Reply:
Tests both the refire logic and the remote state snapshot channel.